nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-reporting-task/src/main/java/org/apache/nifi/reporting/prometheus/PrometheusRecordSink.java

```diff
@@ -62,13 +63,22 @@
     private volatile Map<String, Gauge> gauges;
     private static final CollectorRegistry RECORD_REGISTRY = new CollectorRegistry();
 
+    public static final PropertyDescriptor SSL_CONTEXT = new PropertyDescriptor.Builder()
+            .name("prometheus-reporting-task-ssl-context")
+            .displayName("SSL Context Service")
+            .description("The SSL Context Service to use in order to secure the server. If specified, the server will"
+                    + "accept only HTTPS requests; otherwise, the server will accept only HTTP requests")
+            .required(false)
+            .identifiesControllerService(RestrictedSSLContextService.class)
+            .build();
```

Comment:
Maybe require HTTPS by default, and require explicit override to allow plaintext listening? Idea being, make it harder to misconfigure NiFi --you would have to explicitly shoot yourself in the foot.

Reply:
That's a good point, but this was just moving the existing properties into the existing Prometheus components (since we don't have a dependency on things like RestrictedSSLContextService in the module I added the REST call, I just moved the other common components to a util JAR), changing the default would prevent anyone from running over HTTP. I think that's why these properties default to false everywhere in NiFi, otherwise you must run a secure instance to use the "securable" components.

Reply:
Fair enough... changing all that would be well out of scope for this PR. I think it's something to look into in the future (changing all of NiFi to be secure by default), but not a worry for now.